spatialdata/_constants/_enum.py

```diff
@@ -0,0 +1,66 @@
+from abc import ABC, ABCMeta
+from enum import Enum, EnumMeta
+from functools import wraps
+from typing import Any, Callable, Dict, Tuple, Type
+
+
+class PrettyEnum(Enum):
+    """Enum with a pretty :meth:`__str__` and :meth:`__repr__`."""
+
+    @property
+    def v(self) -> Any:
+        """Alias for :attr`value`."""
+        return self.value
+
+    def __repr__(self) -> str:
+        return f"{self.value!r}"
+
+    def __str__(self) -> str:
+        return f"{self.value!s}"
+
+
+def _pretty_raise_enum(cls: Type["ErrorFormatterABC"], func: Callable[..., Any]) -> Callable[..., Any]:
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any) -> "ErrorFormatterABC":
+        try:
+            return func(*args, **kwargs)  # type: ignore[no-any-return]
+        except ValueError as e:
+            _cls, value, *_ = args
+            e.args = (cls._format(value),)
+            raise e
+
+    if not issubclass(cls, ErrorFormatterABC):
+        raise TypeError(f"Class `{cls}` must be subtype of `ErrorFormatterABC`.")
+    elif not len(cls.__members__):  # type: ignore[attr-defined]
+        # empty enum, for class hierarchy
+        return func
+
+    return wrapper
+
+
+class ABCEnumMeta(EnumMeta, ABCMeta):
+    def __call__(cls, *args: Any, **kwargs: Any) -> Any:
+        if getattr(cls, "__error_format__", None) is None:
+            raise TypeError(f"Can't instantiate class `{cls.__name__}` " f"without `__error_format__` class attribute.")
+        return super().__call__(*args, **kwargs)
+
+    def __new__(cls, clsname: str, superclasses: Tuple[type], attributedict: Dict[str, Any]) -> "ABCEnumMeta":
+        res = super().__new__(cls, clsname, superclasses, attributedict)  # type: ignore[arg-type]
+        res.__new__ = _pretty_raise_enum(res, res.__new__)  # type: ignore[assignment,arg-type]
+        return res
+
+
+class ErrorFormatterABC(ABC):
+    """Mixin class that formats invalid value when constructing an enum."""
+
+    __error_format__ = "Invalid option `{0}` for `{1}`. Valid options are: `{2}`."
+
+    @classmethod
+    def _format(cls, value: Enum) -> str:
+        return cls.__error_format__.format(
+            value, cls.__name__, [m.value for m in cls.__members__.values()]  # type: ignore[attr-defined]
+        )
+
+
+class ModeEnum(str, ErrorFormatterABC, PrettyEnum, metaclass=ABCEnumMeta):
```

Comment:
Not a blocker, but it feels overly complex to use multiple inheritance for an Enum that tracks mode. I think this can be refactored into a more simple structure that just inherits from `PrettyEnum` if the error formatter method is added directly to `PrettyEnum` (i.e., without a mixin) and the `ABCEnumMeta` is part of `PrettyEnum`. If we find there's a need for more granular subclasses, we can move there, but I don't think we want to start out with such complexity.

Reply:
ok, I think it's a nice enum improvement (got it from squidpy) for printing the available enums when input is wrong. I'll try to simplify. It's also a minor part of the code that doesn't really relate to anything else beside enum

Reply:
agree, if you don't mind I'll keep it as is and leave it as a todo for later